Shoko.Server/API/v3/Controllers/TreeController.cs

```diff
@@ -198,47 +211,50 @@ public ActionResult<List<Series>> GetSeriesInFilteredGroup([FromRoute] int filte
         /// </remarks>
         /// <param name="filterID"><see cref="Filter"/> ID</param>
         /// <param name="groupID"><see cref="Group"/> ID</param>
-        /// <param name="recursive">Look for the main series among all the <see cref="Series"/> within the <see cref="Group"/></param>
-        /// <param name="includeMissing">Include <see cref="Series"/> with missing <see cref="Episode"/>s in the search.</param>
         /// <param name="randomImages">Randomise images shown for the main <see cref="Series"/> within the <see cref="Group"/>.</param>
         /// /// <returns></returns>
         [HttpGet("Filter/{filterID}/Group/{groupID}/MainSeries")]
-        public ActionResult<Series> GetMainSeriesInFilteredGroup([FromRoute] int filterID, [FromRoute] int groupID, [FromQuery] bool recursive = false, [FromQuery] bool includeMissing = false, [FromQuery] bool randomImages = false)
+        public ActionResult<Series> GetMainSeriesInFilteredGroup([FromRoute] int filterID, [FromRoute] int groupID, [FromQuery] bool randomImages = false)
         {
             // Return the groups with no group filter applied.
             if (filterID == 0)
-                return GetMainSeriesInGroup(groupID, recursive, includeMissing, randomImages);
+                return GetMainSeriesInGroup(groupID, randomImages);
 
             // Check if the group filter exists.
             var groupFilter = RepoFactory.GroupFilter.GetByID(filterID);
             if (groupFilter == null)
                 return NotFound(FilterController.FilterNotFound);
 
             if (groupFilter.ApplyToSeries != 1)
-                return GetMainSeriesInGroup(groupID, recursive, includeMissing, randomImages);
+                return GetMainSeriesInGroup(groupID, randomImages);
 
             // Check if the group exists.
             var group = RepoFactory.AnimeGroup.GetByID(groupID);
             if (group == null)
                 return NotFound(GroupController.GroupNotFound);
 
-            SVR_AnimeSeries defaultSeries = null;
+            var user = User;
+            if (user.AllowedGroup(group))
+                return Forbid(GroupController.GroupForbiddenForUser);
 
             // Just return early because the every series will be filtered out.
-            if (!groupFilter.SeriesIds.TryGetValue(User.JMMUserID, out var seriesIDs))
+            if (!groupFilter.SeriesIds.TryGetValue(user.JMMUserID, out var seriesIDs))
                 return null;
 
-            // Check if a default series is set.
-            defaultSeries = group.GetDefaultSeries();
-            if (defaultSeries != null && seriesIDs.Contains(defaultSeries.AnimeSeriesID))
-                return new Series(HttpContext, defaultSeries, randomImages);
+            // Return the auto or user selected main series if set.
+            var mainSeries = group.GetMainSeries();
+            if (mainSeries != null)
+                return new Series(HttpContext, mainSeries, randomImages);
```

Comment:
Why are we doing this variable rename pointlessly?

Reply:
Because the value it would retrieve was different.